common/v2/features/PurchaseMembership/PurchaseMembershipStepper.tsx

```diff
@@ -21,38 +26,85 @@ const initialMembershipFlowState = {
+
+  // @ts-ignore Ignore type issue with signing step for now
   const steps: IStepperPath[] = [
     {
-      label: 'Purchase Membership',
+      label: translateRaw('PURCHASE_MEMBERSHIP'),
       component: MembershipPurchaseForm,
-      props: (state => state)(purchaseMembershipFlowState),
-      actions: (formData: ISimpleTxFormFull, cb: any) => handleUserInputFormSubmit(formData, cb)
-    },
-    {
-      label: 'Confirm Transaction',
-      component: ConfirmMembershipPurchase,
-      props: (({ txConfig, membershipSelected }) => ({ txConfig, membershipSelected }))(purchaseMembershipFlowState)
-    },
-    {
-      label: '',
-      component: SignTransaction,
-      props: (({ txConfig }) => ({ txConfig }))(purchaseMembershipFlowState),
-      actions: (payload: ITxReceipt | ISignedTx, cb: any) => handleTxSigned(payload, cb)
+      props: (s => s)(purchaseMembershipFlowState),
+      actions: (formData: MembershipSimpleTxFormFull) => {
+        initWith(
+          () => {
+            const purchaseTx = createPurchaseTx(formData);
+            const approveTx = createApproveTx(formData);
+            return Promise.resolve(
+              isERC20Tx(formData.asset) ? [approveTx, purchaseTx] : [purchaseTx]
+            );
+          },
+          formData.account,
+          formData.account.network
+        );
+        handleUserInputFormSubmit(formData);
+      }
     },
+    ...transactions.flatMap((tx: Required<TxParcel>, idx) => [
+      {
+        label: translateRaw('CONFIRM_TRANSACTION'),
+        backBtnText: translateRaw('PURCHASE_MEMBERSHIP'),
+        component:
+          transactions.length > 1 ? ConfirmMembershipPurchaseMultiTx : ConfirmMembershipPurchase,
+        props: {
+          membershipSelected,
+          account,
+          isSubmitting,
+          transactions,
+          currentTxIdx: idx
+        },
+        actions: () => prepareTx(tx.txRaw)
+      },
+      {
+        label: translateRaw('CONFIRM_TRANSACTION'),
+        backBtnText: translateRaw('CONFIRM_TRANSACTION'),
+        component: account && WALLET_STEPS[account.wallet],
+        props: {
+          network: account && account.network,
+          senderAccount: account,
+          rawTransaction: tx.txRaw
+        },
+        actions: {
+          onSuccess: (payload: ITxHash | ITxSigned) => sendTx(payload)
+        }
+      }
+    ]),
     {
-      label: 'Membership Purchase Receipt',
+      label: translateRaw('PURCHASE_MEMBERSHIP_RECEIPT'),
       component: MembershipPurchaseReceipt,
-      props: (({ txConfig, txReceipt }) => ({ txConfig, txReceipt }))(purchaseMembershipFlowState)
+      props: {
+        account,
+        transactions,
+        membershipSelected
+      }
     }
   ];
+
   return (
     <GeneralStepper
+      onRender={goToNextStep => {
+        useEffect(() => {
+          if (!canYield) return;
+          goToNextStep();
+          stopYield();
+        }, [canYield]);
+      }}
```

Comment:
Still not sure what this is supposed to be after looking at it.

Reply:
It's a way of waiting for the `transactions` to be in state before attempting to navigate to the next step. We do the same in the swap flow. Since `transactions` isn't available before initializing MTX, going to the next step regularly would skip the actual steps of doing the transactions.